vertx-web-client/src/main/java/io/vertx/ext/web/client/cache/CacheManager.java

```diff
@@ -0,0 +1,28 @@
+package io.vertx.ext.web.client.cache;
+
+import io.vertx.core.Handler;
+import io.vertx.ext.web.client.HttpRequest;
+import io.vertx.ext.web.client.HttpResponse;
+
+/**
+ * @author Alexey Soshin
+ */
+public interface CacheManager {
+
+    /**
+     * Clears the cache completely
+     */
+    void flush();
+
+    /**
+     * Attempts to fetch value corresponding to request from cache
+     * @param request       request is used to build cache key
+     * @param hitHandler    this handler is called if key was found in cache
+     *                      to return the value immediately
+     * @param missHandler   in case key is not in cache, you should supply a handler
+     *                      that will be called when the request returns to set the value
+     */
+    void fetch(HttpRequest request,
```

Comment:
I think it would be better to have instead:

```
Future<HttpResponse<Object>> fetch(HttpRequest request)
```

then the fetch method can either return a `Future.succeeded()` or delayed future.

we have something like this in HttpClient redirect policy

Reply:
I think the best we can achieve here is 
```HttpResponse<Object> fetch(HttpRequest request,
               Handler<Handler<HttpResponse<Object>>> missHandler);```

Reply:
actually with decoupling the cache manager code to the interceptor it will be not needed because the cache manager interface will only cache operation and nothing releated to HTTP